src/basic/hashmap.c

```diff
@@ -2135,10 +2135,14 @@ int _hashmap_dump_sorted(HashmapBase *h, void ***ret, size_t *ret_n) {
                 return 0;
         }
 
-        entries = new(struct hashmap_base_entry*, _hashmap_size(h));
+        /* We append one more element than needed so that the resulting array can be used as a strv. We
+         * don't count this entry in the returned size. */
+        entries = new(struct hashmap_base_entry*, _hashmap_size(h) + 1);
         if (!entries)
                 return -ENOMEM;
 
+        entries[_hashmap_size(h)] = NULL;
```

Comment:
i'd move this after the HASHMAP_FOREACH_IDX() below, so that you can just use `n` as index